frameworks/spark/src/main/java/com/criteo/hadoop/garmadon/spark/listener/GarmadonSparkListener.java

```diff
@@ -77,6 +78,43 @@ private void sendExecutorStateEvent(long time, State state, String executorId, S
         this.eventHandler.accept(time, buildOverrideHeader(executorId), executorStateEvent.build());
     }
 
+    private void sendApplicationStateEvent(long time, State state) {
+        SparkConf conf = new SparkConf();
+        String deployMode = conf.get("spark.submit.deployMode", "unknown");
+
+        int driverMemoryMb = (int) conf.getSizeAsMb("spark.driver.memory", "0");
+        int driverMemoryOverheadMb = (int) conf.getSizeAsMb("spark.driver.memoryOverhead", "0");
+        int driverCores = (int) conf.getInt("spark.driver.cores", 0);
+
+        int executorMemoryMb = (int) conf.getSizeAsMb("spark.executor.memory", "0");
+        int executorMemoryOverheadMb = (int) conf.getSizeAsMb("spark.executor.memoryOverhead", "0");
```

Comment:
we should also get the value of spark.executor.memoryOverheadFactor that will be used if memoryOverhead is not set

Reply:
Done